src/wms/src/main/java/org/geoserver/wms/GetLegendGraphicRequest.java

```diff
@@ -134,6 +134,21 @@ public LegendRequest(FeatureType featureType) {
             this.layerName = featureType.getName();
         }
 
+        /**
+         * LegendRequest for a feature type, additional details (title and legend graphic) provided
+         * by MapLayerInfo.
+         *
+         * @param featureType
+         * @param layerName layerName distinct to featureType name
+         */
+        public LegendRequest(FeatureType featureType, Name layerName) {
```

Comment:
Is this change actually related to the GetLegendGraphic on raster ignoring the pre-drawn legend?

Reply:
In case of GridCoverage the featureType Name provided is not the layer name, and without this change the blue square is returned.

Reply:
Ah right, it's still pretending that it's a vector... yeah, I remember now